samples/xaudio/main.c

```diff
@@ -0,0 +1,79 @@
+#include <hal/debug.h>
+#include <hal/video.h>
+#include <hal/xbox.h>
+#include <windows.h>
+#include <string.h>
+#include <hal/audio.h>
+#include <xboxkrnl/xboxkrnl.h>
+#include <assert.h>
+
+#define MIN(x,y) ((x)<(y)?(x):(y))
+#include "nxdk_wav.h"
+
+unsigned int callback_count;
+
+unsigned char *voice_data;
+unsigned int voice_len;
+unsigned int voice_pos;
+
+#define NUM_BUFFERS 2
+const unsigned short buffer_size = 48*1024;
+unsigned char *buffers[NUM_BUFFERS];
+unsigned int current_buf;
+
+static void provide_samples_callback(void *pac97Device, void *data)
+{
+    int is_final = (voice_pos+buffer_size) >= voice_len;
+    int chunk_size = MIN(voice_len-voice_pos, buffer_size);
+
+    memcpy(buffers[current_buf], voice_data+voice_pos, chunk_size);
+    XAudioProvideSamples(buffers[current_buf], chunk_size, is_final);
+
+    if (is_final) {
+        voice_pos = 0;
+    } else {
+        voice_pos = voice_pos+chunk_size;
+    }
+
+    current_buf = (current_buf+1) % NUM_BUFFERS;
+    callback_count++;
+}
+
+int main(void)
+{
+    XVideoSetMode(640, 480, 32, REFRESH_DEFAULT);
+
+    for (int i = 0; i < NUM_BUFFERS; i++) {
+        buffers[i] = MmAllocateContiguousMemoryEx(buffer_size,0,0x3ffb000,0,0x404);
```

Comment:
nit: We use something like `#define MAX_RAM 0x3ffb000` for another sample.

Reply:
Same for your other comment for this line above

Reply:
Done with a FIXME to move this to another file. There's no common spot for it yet. We should refactor all samples that use this constant similarly.